source/basic.tex

```diff
@@ -128,7 +128,7 @@
 specifier (this usage is deprecated; see \ref{depr.static_constexpr}),
 \item
 \indextext{declaration!class name}%
-it is a class name declaration\iref{class.name},
+it is an \grammarterm{elaborated-type-specifier}\iref{class.name},
```

Comment:
There's a category error here; an *elaborated-type-specifier* is not a *declaration*.

[I think the existing wording was attempting to refer to this portion of [class.name]p2: "A *declaration* consisting of *class-key identifier*`;` [...] introduces the class name into the current scope." but generalizing this to apply to other cases in which *elaborated-type-specifier*s introduce *class-name*s seems like a good idea.]

To some extent the problem is that this whole paragraph is bogus: "A declaration is a definition unless" is one big category error. A declaration declares zero or more entities, defines zero or more entities, and introduces zero or more names. But it's wrong to say that a declaration "is" or "is not" a definition.

Perhaps a different structure would help. For instance:

> Each entity declared by a *declaration* is also *defined* by that declaration unless:
> * [...]
> * the entity is introduced by an *elaborated-type-specifier*,
> * [...]
> A declaration is said to be a *definition* of each entity that it defines.

Or perhaps we need to formally divorce the notion of "*declaration*" (the grammar element) and "declaration" (the portion of a *declaration* that declares a specific entity).

Reply:
I like the "different structure" approach, absent the major task of doing the divorce.